src/main/java/io/github/nucleuspowered/nucleus/modules/ticket/data/TicketDataManager.java

```diff
@@ -0,0 +1,294 @@
+import org.apache.commons.lang3.tuple.Triple;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.service.sql.SqlService;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+
+public class TicketDataManager {
+    public static String DATABASE_URL = "jdbc:h2:./nucleus/TicketData";
+
+    public static String[] CREATE_TABLE_QUERIES = {
+            //Create table to store basic ticket data
+            "CREATE TABLE IF NOT EXISTS Tickets(" +
+                    "ID INTEGER PRIMARY KEY AUTO_INCREMENT NOT NULL, " +
+                    "Owner CHAR(36) NOT NULL, " +
+                    "Assignee CHAR(36), " +
+                    "CreationDate TIMESTAMP NOT NULL, " +
+                    "LastUpdateDate TIMESTAMP NOT NULL, " +
+                    "Closed BOOLEAN NOT NULL);",
+            //Create indexes for Tickets table by owner, assignee and closed.
+            "CREATE INDEX IF NOT EXISTS owner ON Tickets(Owner);",
+            "CREATE INDEX IF NOT EXISTS assignee ON Tickets(Assignee);",
+            "CREATE INDEX IF NOT EXISTS closed ON Tickets(Closed);",
+            //Create table to store ticket messages
+            "CREATE TABLE IF NOT EXISTS Ticket_Messages(" +
+                    "ID INTEGER PRIMARY KEY AUTO_INCREMENT NOT NULL, " +
+                    "TicketID INTEGER NOT NULL, " +
+                    "MessageDate TIMESTAMP NOT NULL, " +
+                    "Message TEXT NOT NULL, " +
+                    "CONSTRAINT FK_ID FOREIGN KEY(TicketID) REFERENCES Tickets(ID) ON DELETE CASCADE);",
+            //Create index for Ticket_Messages table by TicketID.
+            "CREATE INDEX IF NOT EXISTS ticketid ON Ticket_Messages(TicketID);"
+    };
+
+    public static String CREATE_TICKET_QUERY = "INSERT INTO Tickets(Owner, Assignee, CreationDate, LastUpdateDate, Closed) VALUES(?, ?, ?, ?, ?);";
+    public static String UPDATE_TICKET_QUERY = "UPDATE Tickets SET Owner = ?, Assignee = ?, CreationDate = ?, LastUpdateDate = ?, Closed = ? WHERE ID = ?;";
+    public static String DELETE_TICKET_QUERY = "DELETE FROM Tickets WHERE ID = ?;";
+
+    public static String CREATE_TICKET_MESSAGE_QUERY = "INSERT INTO Ticket_Messages(TicketID, MessageDate, Message) VALUES(?, ?, ?);";
+    public static String UPDATE_TICKET_MESSAGE_QUERY = "UPDATE Ticket_Messages SET Message = ? WHERE(TicketID = ? AND MessageDate = ?);";
+    public static String DELETE_TICKET_MESSAGE_QUERY = "DELETE FROM Ticket_Messages WHERE(TicketID = ? AND MessageDate = ? AND Message = ?);";
+
+    private NucleusPlugin nucleus;
+    @Inject private TicketConfigAdapter tca;
+    private ExecutorService executor;
+    private TicketDataCache cache;
+
+    @Inject
+    public TicketDataManager(NucleusPlugin nucleus) {
```

Comment:
May as well just inject the `TicketConfigAdapter` here and make it a `private final` variable assigned in the constructor.